core/system/src/mpu/vmpu.c

```diff
@@ -59,16 +59,24 @@ static int vmpu_sanity_checks(void)
     assert(vmpu_bits(0x8001UL) == 16);
     assert(vmpu_bits(1) == 1);
 
+    /* Verify that the known hard-coded symbols are equal to the ones taken from
+     * the host linker script. */
+    assert((uint32_t) __uvisor_config.flash_start == FLASH_ORIGIN);
+    assert((uint32_t) __uvisor_config.flash_offset == FLASH_ORIGIN + RESERVED_FLASH);
+    assert((uint32_t) __uvisor_config.sram_start == SRAM_ORIGIN);
+    assert((uint32_t) __uvisor_config.sram_offset == SRAM_ORIGIN + RESERVED_SRAM);
```

Comment:
Based on your forward looking comment that these _will_ represent the minimum _at some future time_, I was criticizing the asserts as currently written. I'm completely fine with the asserts as is for now, and, as I later realized, changing the asserts to ensure sure there is enough room is a natural thing to do at the _future_ time where we need that behavior.